packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -8,9 +8,36 @@
 import Module from 'module';
 import {join} from 'path';
 import importFresh from 'import-fresh';
-import {LoadContext, Plugin, PluginConfig} from '@docusaurus/types';
+import {
+  LoadContext,
+  Plugin,
+  PluginConfig,
+  ValidationSchema,
+} from '@docusaurus/types';
 import {CONFIG_FILE_NAME} from '../../constants';
 
+function validate<T>(schema: ValidationSchema<T>, options: Partial<T>) {
+  const {error, value} = schema.validate(options, {
```

Comment:
similar to Joi.attempt?

Reply:
Joi.attempt is not avaliable on Schema, so to make it joi indpendent I used this method

Reply:
ok that does not seem a big deal :)